docs/proposals/27-project-group.md

```diff
@@ -0,0 +1,197 @@
+kind: ProjectGroup
+metadata:
+  name: group1
+spec:
+  namespace: gardenprojectgroup-group1
+  owner:
+    apiGroup: rbac.authorization.k8s.io
+    kind: User
+    name: john.doe@example.com
+  members:
+  - apiGroup: rbac.authorization.k8s.io
+    kind: User
+    name: alice.doe@example.com
+    role: admin
+  projects:
+    - Project1
+    - Project2
+```
+
+The main idea behind `ProjectGroup`s is that they, like projects, have their own namespace.
+Every resource that is shareable (at first only `SeedBinding`s) in this namespace, will be shared to all projects in this project group.
+
+This sharing mechanism will be achieved by the `ProjectGroup` controller.
+It will copy the objects from the project group namespace to the namespaces of the projects that belong to the project group.
+
+When the object in the project group namespace is changed, the controller will immediately update the objects in the namespaces of the projects.
+
+To let a user in a project know that such an object is being kept in sync with an object in a project group, a `projectgroup.gardener.cloud/copied-from: <projectgroup-name>` label will be added to it.
+
+This copying feature can become useful for future use cases, for example for sharing infrastructure credentials across different projects, by copying `SecretBinding`s.
+
+It will not be possible to add non-existent project to a project group. This also means that if a project is deleted that is part of a project group, the controller should remove it.
+
+The `.spec.namespace` field in `ProjectGroup` will behave equally as the `Project` `.spec.namespace` field.
+
+##### Permissions
+
+Anybody should be able to create a project group.
+
+But adding a project to a project group is a sensitive operation.
+
+If anyone could add any project to their project group, a bad actor could add a project of a third party to their project group and restrict them from using any seed except the ones bound in the bad actors project group.
+
+To circumvent a scenario like this, a new role, called `project-group-assigner` will be introduced that allows people with this role to add the project to a project group, in which they have the same role.
+
+#### SeedBinding
+
+To enable the general functionality of the two scenarios described in the [Motivation](#motivation) in a singular concept, `SeedBinding`s shall be introduced.
+
+Seed bindings will be considered during the scheduling of a shoot, in a way that if a shoot belongs to a project with a seed binding, only seeds that are targeted by the seed binding will be selected for scheduling.
+
+```yaml
+apiVersion: core.gardener.cloud/v1beta1
+kind: SeedBinding
+metadata:
+  name: privateseed-provider-1
+  namespace: garden-project1
+taintSeed: false
+seedSelector:
+  matchLabels:
+    foo: bar
+
+```
+
+If `.taintSeed` is set to `true`, the specified seed(s) will be tainted, so that no other projects outside the project group can schedule shoots on them.
```

Comment:
What will the taint be? Will the shoot specifications in the project(s) get corresponding tolerations?

Reply:
There is a risk that someone accidentally creates taints for public seeds with such a feature.
Additionally, our seed specs are static at the moment. This feature would change that and so the way we define seeds in our landscapes.
Defining which seed is private and which is public should be part of the landscape definition, but not based on an object in a project or project group.
Thus, I would not implement this feature.

Reply:
An alternative would be to specify an optional toleration to the `SeedBinding` resource. When this is is set, the project will receive the toleration in `.spec.tolerations.[]whitelist` and `.spec.tolerations.[]defaults`.

With this approach, Gardener operators would need to manually set the tolerations to the seeds that should become private and then create the `SeedBinding` with the correlating toleration field.

This would enhance the work of Gardener operators, but decreases accidental mistakes.

I think this approach is also fine, especially when looking at the problems that automatic tainting could produce.
What do you think of this half-automatic approach?

Reply:
Is this feature very useful?

W/o "automatic" tainting, operators need to taint the seeds individually but on the other hand, they already need to put proper labels on the seeds, so that they are selectable, right?